### PR TITLE
CSHARP-2157: Deprecate MongoDB-CR

### PR DESCRIPTION
**Pull request by vincentkam**
Tagged the class as obsolete + added pragmas.

### CODE REVIEW

File: tests/MongoDB.Driver.Tests/MongoCredentialTests.cs [hunk 1]
```diff
@@ -91,10 +95,10 @@ public void TestEquals()
         [Fact]
         public void TestPassword()
         {
-            var credentials = MongoCredential.CreateMongoCRCredential("database", "username", "password");
 #pragma warning disable 618
+            var credentials = MongoCredential.CreateMongoCRCredential("database", "username", "password");
             Assert.Equal("password", credentials.Password);
-#pragma warning restore
```
Comment (vincentkam) [inline tests/MongoDB.Driver.Tests/MongoCredentialTests.cs:97]: 618 was missing so I brought it back.

File: src/MongoDB.Driver/MongoCredential.cs [hunk 2]
```diff
@@ -255,7 +257,7 @@ public static MongoCredential CreateMongoCRCredential(string databaseName, strin
         }
 
         /// <summary>
-        /// Creates a credential used with MONGODB-CR.
+        /// Creates a credential used with MONGODB-X509.
```
Comment (vincentkam) [inline src/MongoDB.Driver/MongoCredential.cs:260]: Fixed what appears to be a typo.

File: src/MongoDB.Driver.Core/Core/Authentication/MongoDBCRAuthenticator.cs [hunk 3]
```diff
@@ -30,7 +30,9 @@ namespace MongoDB.Driver.Core.Authentication
 {
     /// <summary>
     /// A MONGODB-CR authenticator.
+    /// This authenticator was replaced by <see cref="ScramSha1Authenticator"/> in MongoDB 3.0, and is now deprecated.
     /// </summary>
+    [Obsolete("This authenticator was replaced by ScramSha1Authenticator in MongoDB 3.0, and is now deprecated.")]
```
Comment (vincentkam) [inline src/MongoDB.Driver.Core/Core/Authentication/MongoDBCRAuthenticator.cs:35]: Wording taken from https://github.com/jyemin/mongo-java-driver/pull/107/files

**Review by rstam — APPROVED** [review]

File: tests/MongoDB.Driver.Tests/MongoCredentialTests.cs (hunk 1 above)
Comment (rstam) [reply tests/MongoDB.Driver.Tests/MongoCredentialTests.cs:97]: In the past we've tended to restore all warnings, so it was missing 618 on purpose. So you'll find tons of lines like this one in the code base.

But it's not a bad idea to restore JUST the warning you disabled.

I guess the minor risk we were protecting against was an error like this one:

    #pragma warning disable 618
    // ...
    #pragma warning restore 619

**Comment by vincentkam** [comment]
Rebased, squashed, pushed.

**Pull request merged**